src/modules/command-handler/index.js

```diff
@@ -0,0 +1,260 @@
+const Command = require('../command')
+const Await = require('../await')
+const Replacer = require('../replacer')
+
+/**
+ * A class reprsenting the command handler.
+ */
+class CommandHandler {
+  /**
+   * Create a CommandHandler
+   * @param {CommandHandlerData} data The command handler data.
+   */
+  constructor ({ prefix, client, ownerId, knex, replacers = [], commands = [] }) {
+    /**
+     * The prefix of commands.
+     * @type {String}
+     */
+    this._prefix = prefix
+    /**
+     * The Eris Client.
+     * @type {Client}
+     */
+    this._client = client
+    /**
+     * The simple-knex query builder.
+     * @type {QueryBuilder}
+     */
+    this._knex = knex
+    /**
+     * Map of commands.
+     * @type {Map<String, Command>}
+     */
+    this._commands = new Map()
+    /**
+     * An object containing message data used to wait for user response.
+     * @type {Map<String, AwaitData>}
+     */
+    this._awaits = new Map()
+    /**
+     * The ID of the bot owner.
+     * @type {String}
+     */
+    this._ownerId = ownerId
+    /**
+     * Map of replacers.
+     * @type {Map<String, Replacer>}
+     */
+    this._replacers = new Map()
+    // load some commands
+    this.loadCommands(commands)
+    this.loadReplacers(replacers)
+  }
+
+  /**
+   * Load commands.
+   * @param {Command[]|Command} commands The command(s) to load.
+   */
+  loadCommands (commands) {
+    if (commands instanceof Array) {
+      for (let i = 0; i < commands.length; i++) {
+        this._loadCommand(commands[i])
+      }
+    } else {
+      this._loadCommand(commands)
+    }
+  }
+
+  /**
+   * Load replacers.
+   * @param {Replacer[]|Replacer} replacers The replacer(s) to load.
+   */
+  loadReplacers (replacers) { // FIXME: This is redundant with loadCommands
```

Comment:
Just a reminder to fix ;)

Reply:
There's a reason i wrote FIXME and not a fix ;)

Reply:
Just making sure you didn't forget about it ;)

Reply:
~the reason is that im not gonna fix it~

Reply:
Ohhhh, you want me to do it. Ok

Reply:
I'll keep this unresolved so when I look back I'll remember to.